Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -47,96 +44,87 @@ class VirtualMachineSlot: NSObject, ObservableObject {
 
     @MainActor
     func start(launchConfiguration: LaunchConfiguration) async throws {
-
-        self.status = .starting(launchConfiguration)
+        guard isAvailable else {
+            throw Errors.invalidStartState
+        }
 
         do {
-            let virtualMachine = try await launchConfiguration.setupVirtualMachine()
-            virtualMachine.delegate = self
-            self.managedVirtualMachine = ManagedVirtualMachine(machine: virtualMachine, config: launchConfiguration)
-
-            try await virtualMachine.start()
-
-            if launchConfiguration.waitForNetworking {
-                let ipAddress = try await vmManager.ipAddress(forVmWithName: launchConfiguration.handle)
-                Logger.helper.log("Startup complete – IP Address: \(ipAddress.debugDescription)")
-                self.status = .running(launchConfiguration, ipAddress)
-            } else {
-                Logger.helper.log("Startup in progress – skipped waiting for IP address per launch configuration")
-                self.status = .running(launchConfiguration, .any)
+            let mvm = ManagedVirtualMachine(config: launchConfiguration)
+            let startTask = Task { try await mvm.start() }
+            self.status = .starting(mvm, startTask)
+
+            try await startTask.value
+            if startTask.isCancelled {
+                Logger.helper.debug("Start task was cancelled for \(mvm.handle)")
+                throw Errors.vmStartCancelled
             }
+            mvm.machine?.delegate = self
+            self.status = .running(mvm)
```

Comment:
The main objective here is to prevent a race where the slot is mutated from a VM that finished starting. It's a good area to double check.